griptape/loaders/base_text_loader.py

```diff
@@ -32,6 +30,10 @@ class BaseTextLoader(BaseLoader, ABC):
     )
     embedding_driver: Optional[BaseEmbeddingDriver] = field(default=None, kw_only=True)
     encoding: str = field(default="utf-8", kw_only=True)
+    reference: Optional[Reference] = field(default=None, kw_only=True)
```

Comment:
Should this be on `BaseLoader`? Or should `Reference` only live on `TextArtifact`s?

Reply:
Yes but there is more to it. Will start a Slack thread :)